lib/parallel_tests/cli.rb

```diff
@@ -88,6 +88,19 @@ def report_results(test_results)
       results = @runner.find_results(test_results.map { |result| result[:stdout] }*"")
       puts ""
       puts @runner.summarize_results(results)
+      report_failure_rerun_commmand(test_results)
+    end
+
+    def report_failure_rerun_commmand(test_results)
+      failing_sets = test_results.select{|test_result| test_result[:exit_status] != 0 }
+      return unless failing_sets.size > 0
+
+      puts "\n\nTests have failed for a parallel_test group. Use the following command to run the group again:\n\n"
+      failing_sets.each do |failing_set|
+        command = failing_set[:command]
+        command = command.gsub('rspec', "rspec --seed #{failing_set[:seed]}") if failing_set[:seed]
```

Comment:
Do all the test frameworks allow a seed argument? I'll check that adding it
to the end works for rspec at least...and then change it if it works...
On 3 Oct 2015 6:14 am, "Michael Grosser" <notifications@github.com> wrote:

> In lib/parallel_tests/cli.rb
> <https://github.com/grosser/parallel_tests/pull/448#discussion_r41084066>:
>
> > @@ -88,6 +88,19 @@ def report_results(test_results)
> >        results = @runner.find_results(test_results.map { |result| result[:stdout] }*"")
> >        puts ""
> >        puts @runner.summarize_results(results)
> > +      report_failure_rerun_commmand(test_results)
> > +    end
> > +
> > +    def report_failure_rerun_commmand(test_results)
> > +      failing_sets = test_results.select{|test_result| test_result[:exit_status] != 0 }
> > +      return unless failing_sets.size > 0
> > +
> > +      puts "\n\nTests have failed for a parallel_test group. Use the following command to run the group again:\n\n"
> > +      failing_sets.each do |failing_set|
> > +        command = failing_set[:command]
> > +        command = command.gsub('rspec', "rspec --seed #{failing_set[:seed]}") if failing_set[:seed]
>
> can we append that to the command so it works for everything ?
>
> —
> Reply to this email directly or view it on GitHub
> <https://github.com/grosser/parallel_tests/pull/448/files#r41084066>.
>